repairnator/repairnator-pipeline/src/main/java/fr/inria/spirals/repairnator/process/step/gatherinfocontract/GatherTestInformation.java

```diff
@@ -34,23 +35,16 @@
     private Set<FailureLocation> failureLocations;
     private Set<String> failureNames;
     private String failingModulePath;
+    private boolean skipSettingStatusInformation;
 
     private ContractForGatherTestInformation contract;
 
-    public GatherTestInformation(ProjectInspector inspector, ContractForGatherTestInformation contract) {
+    public GatherTestInformation(ProjectInspector inspector, ContractForGatherTestInformation contract, boolean skipSettingStatusInformation) {
```

Comment:
I agreed with what you did: `skipSettingStatusInformation` must be false when `BuildShouldFail`, because this is the scenario both of us want to save information. I don't see any better way to implement it now, but what I meant is that we should put some javadoc in the code explaining this correlation.